llm/tests/compare_hf_v_mosaic_gpt.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2022 MosaicML Benchmarks authors
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import warnings
+
+import torch
+from omegaconf import OmegaConf as om
+from composer.utils import reproducibility
+
+from main import build_composer_model
+
+
+def test_compare_hf_v_mosaic_gpt():
+    warnings.filterwarnings(action='ignore', message='Torchmetrics v0.9 introduced a new argument class property')
+    device = 'cuda'
+
+    # get 125m config
+    with open('yamls/mosaic_gpt/125m.yaml') as f:
+        cfg = om.load(f)
+
+    # modify cfg for HF GPT2 compatibility
+    cfg.max_seq_len = 1024  # GPT2 seq len (GPT3 uses 2048)
+    cfg.model.max_seq_len = cfg.max_seq_len
+    cfg.train_loader.max_seq_len = cfg.max_seq_len
+    cfg.eval_loader.max_seq_len = cfg.max_seq_len
+    cfg.model.device = device
+    cfg.precision = 'fp16'
```

Comment:
I think this should be `amp` rather than `fp16` (silly Composer convention), but I also don't think it will affect anything as you are running the forwards pass yourself.

Reply:
deprecated